block/proposals/update.go

```diff
@@ -0,0 +1,105 @@
+package proposals
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/skip-mev/block-sdk/block/utils"
+)
+
+// UpdateProposal updates the proposal with the given transactions and lane limits. There are a
+// few invarients that are checked:
+//  1. The total size of the proposal must be less than the maximum number of bytes allowed.
+//  2. The total size of the partial proposal must be less than the maximum number of bytes allowed for
+//     the lane.
+//  3. The total gas limit of the proposal must be less than the maximum gas limit allowed.
+//  4. The total gas limit of the partial proposal must be less than the maximum gas limit allowed for
+//     the lane.
+//  5. The lane must not have already prepared a partial proposal.
+//  6. The transaction must not already be in the proposal.
+func (p *Proposal) UpdateProposal(lane string, partialProposal []sdk.Tx, limit LaneLimits) error {
```

Comment:
Why would the caller pass in their LaneLimits? Why wouldn't the proposal j get them here from the lane-string? I.e if the purpose of this abstraction is to have devs only aware of adding txs to proposals, does it not make more sense to j hide the notion of a `LaneLimit` entirely?

Reply:
yea i agree, will update